### ISSUE TITLE
nice to have `find_near_duplicates()` in image utilities

### ISSUE THREAD

**Issue by zzsi** [kungfuai/kaishi#1]
Similar to `find_duplicates` which uses md5 hash, sometimes we also want to find near duplicates generated by:

- resizing a photo and then keeping multiple resolutions of the same image
- extracting video frames that are very close in timestamp, or no events happening in between

This can be done by perceptual hash (e.g. https://github.com/jgraving/imagehash), or nearest neighbor + pretrained embedding.

The goal is to reduce train-test data leakage: duplicate or near duplicate examples sitting in both training and test set.
